pkg/stacks/unpack.go

```diff
@@ -220,6 +229,18 @@ func (sp *StackPackage) IsNamespaced() bool {
 func (sp *StackPackage) SetApp(app v1alpha1.AppMetadataSpec) {
 	app.DeepCopyInto(&sp.Stack.Spec.AppMetadataSpec)
 
+	if sp.Stack.Spec.AppMetadataSpec.Version == "" {
+		iv := os.Getenv(StackImageEnv)
```

Comment:
It will likely be a pain, but did you look for ways to plumb `StackImageEnv` through `StackPackage` and other parent types?

I'm not blocking on this, just curious.  The precedence is in your favor with `os.Getenv` already being used in `pkg/stacks/executorinfo.go` and `pkg/stacks/pod.go`. Ideally we would only mess with environment variables in main.go, via kingpin.

Reply:
I did not, because of the precedents, but I agree with the principle of not touching environment variables down in the code. Yeah, ideally we'd pull them out in main and pass around some sort of configuration instead.